core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -17,63 +17,103 @@
    */
-  public static <T> T call(RpcCallable<T> callable) throws AlluxioTException {
+  public static <T> T call(Logger logger, RpcCallable<T> callable) throws AlluxioTException {
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      logger.debug("Exit (Error): {}", callable, e);
       throw e.toThrift();
     } catch (Exception e) {
-      LOG.error("Unexpected error running rpc", e);
+      logger.error("Exit (Error): {}", callable, e);
       throw new UnexpectedAlluxioException(e).toThrift();
     }
   }
 
   /**
    * Calls the given {@link RpcCallableThrowsIOException} and handles any exceptions thrown.
    *
+   * @param logger the logger to use for this call
    * @param callable the callable to call
    * @param <T> the return type of the callable
    * @return the return value from calling the callable
    * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
    * @throws ThriftIOException if the callable throws an IOException
    */
-  public static <T> T call(RpcCallableThrowsIOException<T> callable)
+  public static <T> T call(Logger logger, RpcCallableThrowsIOException<T> callable)
       throws AlluxioTException, ThriftIOException {
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      logger.debug("Exit (Error): {}", callable, e);
       throw e.toThrift();
     } catch (IOException e) {
-      LOG.warn("I/O error when running rpc", e);
+      logger.debug("Exit (Error): {}", callable, e);
       throw new ThriftIOException(e.getMessage());
     } catch (Exception e) {
-      LOG.error("Unexpected error running rpc", e);
+      logger.error("Exit (Error): {}", callable, e);
       throw new UnexpectedAlluxioException(e).toThrift();
     }
   }
 
   /**
-   * An interface representing a callable which can only throw Alluxio exceptions.
+   * Calls the given {@link RpcCallable} and handles any exceptions thrown. The callable should
+   * implement a toString with the following format: "CallName: arg1=value1, arg2=value2,...".
+   * The toString will be used to log enter and exit information with debug logging is enabled.
+   *
+   * @param logger the logger to use for this call
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   */
+  public static <T> T callAndLog(Logger logger, RpcCallable<T> callable) throws AlluxioTException {
+    logger.debug("Enter: {}", callable);
+    T ret = call(logger, callable);
+    logger.debug("Exit (OK): {}", callable);
+    return ret;
+  }
+
+  /**
+   * Calls the given {@link RpcCallableThrowsIOException} and handles any exceptions thrown. The
```

Comment:
the javadoc for the format string seems to be duplicated here and in the RpcCallable interface ... I feel like we only need it in one of those places